mobile/src/main/java/ch/epfl/sdp/mobile/ui/setting/SettingScreen.kt

```diff
@@ -0,0 +1,130 @@
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.state.toColor
+import ch.epfl.sdp.mobile.ui.profile.*
+
+/**
+ * Main component of the ProfileScreen that groups ProfileHeader and list of Matches
+ * @param state state of the ProfileScreen
+ * @param modifier the [Modifier] for this composable.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun SettingsScreen(
+    state: SettingScreenState,
+    modifier: Modifier = Modifier,
+) {
+  val lazyColumnState = rememberLazyListState()
+  val tabBarState = rememberSettingTabBarState(state.pastGamesCount, state.puzzlesCount)
+  val targetElevation = if (lazyColumnState.firstVisibleItemIndex >= 1) 4.dp else 0.dp
+  val elevation by animateDpAsState(targetElevation)
+
+  UserScreen(
+      Header = {
+        SettingHeader(
+            state = state,
+            modifier = Modifier.padding(vertical = 16.dp),
+        )
+      },
+      ProfileTabBar = {
+        SettingTabBar(
+            state = tabBarState,
+            modifier = Modifier.fillMaxWidth(),
+            elevation = elevation,
+        )
+      },
+      matches = state.matches,
+      lazyColumnState = lazyColumnState,
+      modifier = modifier)
+}
+
+/**
+ * Composes the settings header given the profile [SettingScreenState]. Displays also the
+ * ProfilePicture,
+ *
+ * SettingsButton, name and email of the user's profile
+ *
+ * @param state state of profile screen
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+fun SettingHeader(state: SettingScreenState, modifier: Modifier = Modifier) {
+
+  Column(
+      modifier = modifier,
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.spacedBy(16.dp),
+  ) {
+    SettingPicture(state)
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+      Text(state.name, style = MaterialTheme.typography.h5)
+      Text(state.email, style = MaterialTheme.typography.subtitle2)
+    }
+    SettingsButton(onClick = state::onSettingsClick)
+  }
+}
+
+/**
+ * Composes the settings picture given its [state]
+ * @param state state of setting screen
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+fun SettingPicture(
+    state: SettingScreenState,
+    modifier: Modifier = Modifier,
+) {
+  Box(
+      modifier = modifier.size(118.dp).background(state.backgroundColor.toColor(), CircleShape),
+      contentAlignment = Alignment.Center,
+  ) {
+    Text(state.emoji, style = MaterialTheme.typography.h3)
+    IconButton(
+        onClick = state::onEditClick,
+        modifier =
+            Modifier.align(Alignment.BottomEnd)
+                .shadow(2.dp, CircleShape)
+                .background(MaterialTheme.colors.surface, CircleShape)
+                .border(2.dp, MaterialTheme.colors.primary, CircleShape)
+                .size(40.dp),
+    ) { Icon(Icons.Default.Edit, "Edit profile icon") }

```

Comment:
Prefer using a value from `LocalizedStrings` if you provide a content description, rather than a hardcoded string literal.